plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -129,66 +131,97 @@ public Network design(NetworkOffering offering, DeploymentPlan plan, Network use
+                    network.getDataCenterId(), physicalNetworkId, offering.getRedundantRouter());
+            if (network.getGateway() != null) {
+                implemented.setGateway(network.getGateway());
+            }
+            if (network.getCidr() != null) {
+                implemented.setCidr(network.getCidr());
+            }
+            Collection<String[]> ipAddressRanges = new ArrayList<String[]>();
+            String virtualRouterIp = getVirtualRouterIP(network, ipAddressRanges);
+            String networkUuid = implemented.getUuid();
+            String tenantId = context.getDomain().getName() + "-" + context.getAccount().getAccountId();
+            String broadcastUriStr = networkUuid + "/" + virtualRouterIp;
+            implemented.setBroadcastUri(Networks.BroadcastDomainType.Vsp.toUri(broadcastUriStr));
+            implemented.setBroadcastDomainType(Networks.BroadcastDomainType.Vsp);
+            //Check if the network is associated to a VPC
+            Long vpcId = network.getVpcId();
+            boolean isVpc = (vpcId != null);
+            //Check owner of the Network
+            Domain networksDomain = _domainDao.findById(network.getDomainId());
+            //Get the Account details and find the type
+            AccountVO networksAccount = _accountDao.findById(network.getAccountId());
+            if (networksAccount.getType() == Account.ACCOUNT_TYPE_PROJECT) {
+                String errorMessage = "CS project support is not yet implemented in NuageVsp";
+                s_logger.debug(errorMessage);
+                throw new InsufficientVirtualNetworkCapacityException(errorMessage, Account.class, network.getAccountId());
+            }
+            String vpcName = null;
+            String vpcUuid = null;
+            String preConfiguredDomainTemplateName = null;
+            boolean isL3Network = false, isSharedNetwork = false;
+            if (isVpc) {
+                Vpc vpcObj = _vpcDao.findById(vpcId);
+                vpcName = vpcObj.getName();
+                vpcUuid = vpcObj.getUuid();
+                preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspVpcDomainTemplateName.key());
+            } else {
+                if (offering.getGuestType() == GuestType.Shared) {
+                    isL3Network = true;
+                    isSharedNetwork = true;
+                    preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspSharedNetworkDomainTemplateName.key());
+
+                    ipAddressRanges = new ArrayList<String[]>();
+                    List<VlanVO> vlans = _vlanDao.listVlansByNetworkId(network.getId());
+                    for (VlanVO vlan : vlans) {
+                        boolean isIpv4 = StringUtils.isNotBlank(vlan.getIpRange());
+                        String[] range = isIpv4 ? vlan.getIpRange().split("-") : vlan.getIp6Range().split("-");
```

Comment:
The ``spilt`` calls are identical.  Extract the ternary to statement to retrieve the raw range and consolidate the split calls to DRY out the code.